matchbox_socket/src/webrtc_socket/mod.rs

```diff
@@ -15,22 +15,27 @@ pub(crate) use socket::MessageLoopChannels;
 pub use socket::{ChannelConfig, PeerState, RtcIceServerConfig, WebRtcSocket, WebRtcSocketConfig};
 use std::pin::Pin;
 
-use self::error::SignallingError;
-
 cfg_if! {
     if #[cfg(target_arch = "wasm32")] {
         mod wasm;
         type UseMessenger = wasm::WasmMessenger;
         type UseSignaller = wasm::WasmSignaller;
-        type MessageLoopFuture = Pin<Box<dyn Future<Output = Result<(), Error>>>>;
+        /// A future which runs the message loop for the socket and completes
+        /// when the socket closes or disconnects
+        pub type MessageLoopFuture = Pin<Box<dyn Future<Output = Result<(), Error>>>>;
     } else {
         mod native;
         type UseMessenger = native::NativeMessenger;
         type UseSignaller = native::NativeSignaller;
-        type MessageLoopFuture = Pin<Box<dyn Future<Output = Result<(), Error>> + Send>>;
+        /// A future which runs the message loop for the socket and completes
+        /// when the socket closes or disconnects
+        pub type MessageLoopFuture = Pin<Box<dyn Future<Output = Result<(), Error>> + Send>>;
```

Comment:
I wish there was a way not to have platform-specific versions of this...

Reply:
Agreed. It just has to do with WASM, right? since WASM has no threading?

I wonder when/if that becomes stable (threading), or if there's a solution others use we haven't seen.

Reply:
Honestly, I can't remember why, maybe it's worth revisiting